tritium-metrics/src/test/java/com/palantir/tritium/metrics/MetricRegistriesTest.java

```diff
@@ -394,6 +396,42 @@ public void problematicTimestamps() {
         assertParsesTimestamp("2019-03-29T23:38:51.092Z");
     }
 
+    @Test
+    public void codahale_registry_histogram_count_should_monotonically_increase_after_window()
+            throws InterruptedException {
+        MetricRegistry registry = new MetricRegistryWithReservoirs(
+                () -> new SlidingTimeWindowArrayReservoir(1, TimeUnit.MILLISECONDS));
+
+        Histogram histogram = registry.histogram("histogram");
+        histogram.update(20);
+        histogram.update(20);
+        histogram.update(20);
+        histogram.update(20);
+
+        assertThat(histogram.getCount()).isEqualTo(4);
+
+        Thread.sleep(1);
+
+        assertThat(histogram.getCount()).isEqualTo(4);
+    }
+
+    @Test
+    public void tagged_registry_histogram_count_should_monotonically_increase_after_window()
+            throws InterruptedException {
+        SlidingWindowTaggedMetricRegistry registry = new SlidingWindowTaggedMetricRegistry(1, TimeUnit.MILLISECONDS);
+        Histogram histogram = registry.histogram(MetricName.builder().safeName("histogram").build());
+        histogram.update(20);
+        histogram.update(20);
+        histogram.update(20);
+        histogram.update(20);
+
+        assertThat(histogram.getCount()).isEqualTo(4);
+
+        Thread.sleep(1);
+
+        assertThat(histogram.getCount()).isEqualTo(4);
+    }
```

Comment:
happy to delete these Thread.sleep tests, just wanted to demonstrate that things were not consistent before and now they are.

Reply:
we might want a visible for test version of `SlidingWindowTaggedMetricRegistry` constructor that takes a `Clock` to simplify this kind of test